analytics.go

```diff
@@ -97,6 +94,16 @@ func NewWithConfig(writeKey string, config Config) (cli Client, err error) {
 	return
 }
 
+func newHttpClient(transport http.RoundTripper) http.Client {
```

Comment:
Go conventions are to name `make-` functions returning values (`new-` is for functions returning pointers).